### PR TITLE
added from_uc script

### PR DESCRIPTION
**Pull request by gregcaporaso**
~~**NOT READY FOR MERGE**~~

basic unit tests added, more extensive tests already exist for the parser that contains most of the functionality.

### CODE REVIEW

File: biom/cli/uc_processor.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    $ biom from-uc -i in.uc -o out.biom --rep-set-fp rep-set.fna
+
+    """
+    input_f = open(input_fp, 'U')
+    if rep_set_fp is not None:
+        rep_set_f = open(rep_set_fp, 'U')
+    else:
+        rep_set_f = None
+    table = _from_uc(input_f, rep_set_f)
+    write_biom_table(table, 'hdf5', output_fp)
```
Comment (wasade) [inline biom/cli/uc_processor.py]: only biom 2.x supported?

Comment (wasade) [inline biom/cli/uc_processor.py]: if so, is `HAVE_H5PY` respected?

File: biom/cli/uc_processor.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        line = line.strip()
+        if line.startswith('>'):
+            try:
+                obs_id, seq_id = line.split()[:2]
+            except ValueError:
+                raise ValueError('Sequence identifiers in fasta file '
+                                 'must contain at least two space-'
+                                 'separated fields.')
+            result[seq_id] = obs_id[1:]
+        else:
```
Comment (wasade) [inline biom/cli/uc_processor.py]: `else` isn't needed

File: biom/cli/uc_processor.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    else:
+        rep_set_f = None
+    table = _from_uc(input_f, rep_set_f)
+    write_biom_table(table, 'hdf5', output_fp)
+
+
+def _id_map_from_fasta(fasta_lines):
+    result = {}
+    for line in fasta_lines:
+        line = line.strip()
```
Comment (wasade) [inline biom/cli/uc_processor.py]: this isn't needed since only the obs_id is ever retained

File: tests/test_cli/test_uc_processor.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+    def setUp(self):
+        """Set up data for use in unit tests."""
+        self.cmd = _from_uc
+        self.uc_minimal = uc_minimal.split('\n')
+        self.uc = uc.split('\n')
+        self.rep_set = rep_set.split('\n')
+        self.rep_set_no_mapping = rep_set_no_mapping.split('\n')
+        self.rep_set_missing_id = rep_set_missing_id.split('\n')
+        # with tempfile.NamedTemporaryFile('w') as fh:
```
Comment (wasade) [inline tests/test_cli/test_uc_processor.py]: delete please

**Comment by wasade** [comment]
few small comments, :+1: otherwise

File: biom/cli/uc_processor.py (hunk 2 above)
Comment (gregcaporaso) [inline biom/cli/uc_processor.py]: I know, it's stylistic. I like if/elif to always have elses.

File: biom/cli/uc_processor.py (hunk 1 above)
Comment (gregcaporaso) [inline biom/cli/uc_processor.py]: Yep to both (see ``write_biom_table``), can always convert. I would like to use the same writing approach as the other scripts.

File: biom/cli/uc_processor.py (hunk 3 above)
Comment (gregcaporaso) [inline biom/cli/uc_processor.py]: fixed in #675.

**Comment by gregcaporaso** [comment]
Thanks! Closing in favor of #675. Addressed the ``split`` comment there.

**Pull request closed without merging**